homework/ETL.py

```diff
@@ -0,0 +1,106 @@
+import psycopg2
+import pandas as pd
+from sqlalchemy import create_engine
+from datetime import datetime
+
+#################
+#### Extract ####
+#################
+
+
+#Connect to the data base using the connector from psycopg2
+conn = psycopg2.connect(database = "extracts", user = "postgres", host = "localhost")
+#Create a cursor
+cur = conn.cursor()
+
+#Define function for extraction
+def extract(table):
+	"""This functions gets as an input the name of a table in Postgres and 
+		returns as an output the content of the table as a Pandas DataFrame"""
+	#Get columns names
+	sql_names = """SELECT column_name FROM information_schema.columns WHERE table_name = %s;"""
+	cur.execute(sql_names,(table,))
+	names = cur.fetchall()
+	col_names = []
+	for n in names: col_names.append(n[0]) 
+	#Get content
+	sql_content = """SELECT * FROM %s;"""
+	cur.execute(sql_content % (table,))
+	content = cur.fetchall()
+	return(pd.DataFrame(content, columns = col_names))
+
+
+orders_df = extract('orders')
+products_df = extract('products')
+offices_df = extract('offices')
+employees_df = extract('employees')
+customers_df = extract('customers')
+orders_consolidated_df = extract('orders_con')
+
+
+###################
+#### Transform ####
+###################
+
+
+#### Fact Table
+
+#Create order code
+orders_df['order_code'] = orders_df.order_number.map(str) + orders_df.order_line_number.map(str)
+#Check uniqueness
+orders_df.order_code.nunique() == len(orders_df)
```

Comment:
This is a great thing to check, but: 

1. Can you let the database do it? 
2. What would happen here if the length wasn't the same? (hint: You should probably throw an error!)